iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/CmdTreeImpl.java

```diff
@@ -60,17 +61,32 @@ public Kind getKind() {
   }
 
   @Override
+  @CheckForNull
   public ExecFormTree execForm() {
     return execForm;
   }
 
+  @Override
+  @CheckForNull
+  public ShellFormTree shellForm() {
+    return shellForm;
+  }
+
   @Override
   public List<SyntaxToken> cmdArguments() {
     List<SyntaxToken> result = new ArrayList<>();
-    SeparatedList<ExecFormLiteralTree> literals = execForm.literals();
-    if (literals != null) {
-      for (ExecFormLiteralTree element : literals.elements()) {
-        result.add(element.value());
+    if (execForm != null) {
+      SeparatedList<ExecFormLiteralTree> literals = execForm.literals();
+      if (literals != null) {
+        for (ExecFormLiteralTree element : literals.elements()) {
+          result.add(element.value());
+        }
+      }
+    }
+    if (shellForm != null) {
+      List<SyntaxToken> literals = shellForm.literals();
+      if (literals != null) {
+        result.addAll(literals);
       }
```

Comment:
I would tend to skip the second check for null `literals != null` as the method always return an arrays and don't has a Nullable/CheckForNull annotation.
Same for execForm, WDYT ?

Reply:
ok